### ISSUE TITLE
Sup, also project status?

### ISSUE THREAD

**Issue by resttime** [R-a-dio/R-a-dio-Amazing-Android-App#2]
Got a notification from this project. Although I don't use Android anymore, I might be able to help pitch in during my spare time. I feel bad that the previous app I worked on never deployed and sank into all kinds of code hell.

Has there been any updates regarding an API for search/request song? I've also acquired a bit of web programming experience so I could help with that as well.

**Comment by CaptainJet** [comment]
Assigned @Kethsar to issue: He pinged everyone haha.

**Comment by KilimIG** [comment]
hi rest!

we're doing pretty good right now, we'll be finished with this in a few days' time; if that, however i can 100% see the code being filled to the brim and memory leaking being an issue (maybe? idk none of us no what we're doing)

any help would, well, help!

**Comment by resttime** [comment]
# Some Random List of Features

## Background Music 
Playing music without a MainActivity requires an Android service and it will start to complicate things. This would be a fairly high priority after the first release since it will change how components will interact with one another.

## Stopping music during calls
I think this was definitely very high.

## Homescreen widget
Rice

## Notifications
Stuff like DJ has changed or YOUR REQUESTED SONG IS PLAYINGH AHRAJHRJ AHRA or YOUR FAVORITE SONG IS PLAYING

## Notification bar
Ricing widget in that pull down bar to control the player.

## Searches
I don't know whether there has been API changes/additions. This might require scraping the website's search page. There used to be a RESTFUL API in the past but it was dropped after moving to the newer website. It used to allow autocompletion when we searched in the old app.

## Requests
I don't know about whether there's been API changes/additions but the way that the current website does it is by grabbing the **id** of a song when searching and **POST**ing it to **/request/** with a **CSRF token**:

```javascript
 $(".request-button").click(function (e) {
     e.preventDefault();
     var id = $(this).val(),
	 token = $(this).closest("form").find("input[name='_token']").val();
     $.ajax({
	 url: "/request/" + id,
	 type: "POST",
	 data: { _token: token },
	 dataType: "json",
	 success: function (data) {
	     var result;
             
	     if (data.error) {
		 result = data.error;
	     } else if (data.success) {
		 result = data.success;
	     } else {
		 console.log(data);
	     }
             
	     $("#requests .modal-body").text(result);
	     $("#requests").modal();
	     
	 }
     });
```
Basically we need a **token** that is provided to us by the server and the **id** of our song to make a request with this api. Unless the API has changed, this might also require scraping the website's search page.

## Favorites
This would require new API unless kept local I suppose.

## Settings
For whatever

**Comment by CaptainJet** [comment]
Background Music, Search/Request, Notification Control we were already planning to implement, the others seem like they can wait.

On the note of request/search, the site has api for checking they can request at https://r-a-d.io/api/can-request and an api for searching at https://r-a-d.io/api/search/<thing>. There's one for requesting too but idr what it is.

Feel free to fork the project and put in a pull request if you have time to implement any of these yourself.

**Comment by KilimIG** [comment]
oh baby that list is great thanks

**Comment by resttime** [comment]
@CaptainJet 
Alright, I'll work on it in like a few days once I get access to my main machine. 

I've confirmed with others that there's still no other endpoint for requesting except through the website. The HTML scraping is basically a last resort kinda thing I figured while I was waiting back then, definitely not good practice. I'll see if I can facilitate/implement the request endpoint.

@KilimIG
np

**Comment by CaptainJet** [comment]
i'm just gonna edit the list post whenever we finish something

**Comment by KilimIG** [comment]
some more stuff we should add:

**EASY**

drawerbutton (homescreen.xml) should bring up the slider on tap

(ties to the request below) have the text below the search bar return the amount of queries, only if we decide to extend searches

a way to clear search results when searching empty?

after returning a search query and hitting the search bar again, it should delete whats in the container

**MEDIUM**

adding "pages" to requests that return more than 20 strings (once you scroll down to the bottom of a "page" it will display 20 more requests)

**Comment by KilimIG** [comment]
also the whole pause on call whopps

**Comment by CaptainJet** [comment]
Added phone call muting/unmuting and opening the drawer by pressing buttons.

**Comment by Kethsar** [comment]
We should really split each potential thing to add into it's own 'Issue' so we can keep track of them better.

On that note, I have another to add. Update the notification with the song title and artist whenever a new song starts. That way a user can see what song is playing from their lock screen instead of having to open the app.

**Comment by KilimIG** [comment]
closing this for now we can pull from it later
